include/algorithms/synthesis/dd_synthesis.hpp

```diff
@@ -6,4 +6,29 @@ namespace syrec {
 
     auto buildDD(const TruthTable& tt, std::unique_ptr<dd::Package<>>& dd) -> dd::mEdge;
 
+    struct DDSynthesis {
```

Comment:
This should probably be a class instead of a struct in order to only expose functions that are intended to be used publicly.

Furthermore it should probably be called DDSynthesizer or similar.